node/silkworm/downloader/internals/db_tx.hpp

```diff
@@ -202,6 +203,49 @@ class Db::ReadOnlyAccess::Tx {
     }
 
     BlockNum read_stage_progress(const char* stage_name) { return db::stages::read_stage_progress(txn, stage_name); }
+
+    // see Erigon's HeadersUnwind method for the implementation
+    std::tuple<BlockNum, Hash> header_with_biggest_td(const std::set<Hash>* bad_headers = nullptr) {
+        BlockNum max_block_num = 0;
+        Hash max_hash;
+        BigInt max_td = 0;
+
+        auto td_cursor = db::open_cursor(txn, db::table::kDifficulty);
+
+        auto find_max = [bad_headers, &max_block_num, &max_hash, &max_td](mdbx::cursor&,
+                                                                          mdbx::cursor::move_result& result) -> bool {
+            ByteView key = db::from_slice(result.key);
+            ByteView value = db::from_slice(result.value);
+
+            if (key.size() != 40) {
+                throw std::logic_error("key in td table has to be 40 bytes long: " + result.key.as_hex_string());
+            }
+
+            Hash hash{key.substr(8)};
+            ByteView block_num = key.substr(0, 8);
```

Comment:
Semantically I'd prefer to consume key view early. I think is neater and reflects the composition of the key depicted above. 
```suggestion
            auto block_num{endian::load_big_u64(key.data())};
            key.remove_prefix(sizeof(block_num));
            Hash hash{key};
```

Eventually change below `max_block_num = block_num`

Reply:
I think the current code is more readable, it clearly shows that the key is divided in two parts. Maybe I need to substitute 8 with sizeof(block_num)

Reply:
I agree with Mike. `sizeof(block_num)` instead of 8 probably helps readability, while whether `remove_prefix` does so is moot.